Sort out skipped trashbin integration tests

phil-davis opened the pull request:
## Description
1) tag the existing skipped tests with issue-23151
2) add a new step definition that will wait and delete, to work around the timing issue
3) use that step definition in new scenarios that demonstrate the good working behaviour

## Related Issue
#23151 

## Motivation and Context
I noticed 2 trashbin tests that were skipped, while refactoring integration tests. 

I tried the tests, and they pass locally for me, they look decent and I tried combinations of what they do to make sure they would really fail if things are not right. But they fail when run in CI. It turns out they fail when run quickly - if multiple files with the same name are deleted in the same second then there is a naming conflict in the trashbin.

The automated tests can have 1 scenario that work around the timing bug/issue and tests/demonstrates the other correct behaviour of putting files with the same name from different folders into the trashbin. Then the existing scenario with the timing bug/issue can remain skipped, but also be tagged with the issue number so people can easily follow it up in future.

## How Has This Been Tested?
Run the skipped tests locally - they pass
Unskip the tests to run them in CI - they fail there, so we know they are "good" example failing cases

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

phil-davis committed:
Modify trashbin integration tests for issue 23151

phil-davis commented:
These were skipped in https://github.com/owncloud/core/pull/28111
But (assuming CI passes here) they work now.

phil-davis commented:
Hmmm - those tests fail on Jenkins, but pass for me locally. So there is something about the Jenkins environment/timing and the details of this test that needs looking into.

PVince81 commented:
@phil-davis please note that deleted files get a file name like "test.txt.d1234567" where the number is a timestamp. Maybe if it deletes too fast the timestamp is exactly the same so we bump into this issue: https://github.com/owncloud/core/issues/23151

phil-davis commented:
Thanks for the hint - I will get back to this soon...

phil-davis commented:
@SergioBertolinSG @individual-it @PVince81 I have updated the code and squashed.
This should now keep the failing scenarios, better tagging them against the relevant issue, and provide new scenarios that at least test/demonstrate the good behaviour that is there.

Please review again.

phil-davis commented:
Backport stable10 #23151
